spec/bolt/executor_spec.rb

```diff
@@ -310,6 +310,44 @@ def mock_node_results(_run_as = nil)
     end
   end
 
+  context 'waiting until targets are available' do
+    it 'waits on all nodes' do
+      node_results.each do |target, _|
+        expect(ssh)
+          .to receive(:connected?)
+          .with(target)
+          .and_return(true)
+      end
+
+      results = executor.wait_until_available(targets)
+      results.each do |result|
+        expect(result).to be_instance_of(Bolt::Result)
+      end
+    end
+
+    it 'errors after timeout' do
+      allow(ssh).to receive(:connected?).and_return(false)
+
+      results = executor.wait_until_available(targets, wait_time: 0, retry_interval: 0)
```

Comment:
Might also be worth adding a test to ensure that retry_interval and wait_time are respected. wait_time would be easy to test, retry_interval might be trickier...you could maybe count how many times `Time.now` is called?

Reply:
I can work on it. Getting the `wait` tests reliable has proved tricky.

Reply:
Added a new test and improved existing ones around `Time.now`.